### PR TITLE
Added simple QoL fixes for the search filter and more validation for nickname

### PR DESCRIPTION
**Pull request by YingLiang2**
* Filter now starts filtering with 1 input character in the search field.
* Nickname filtering now includes a max of 15 chars, and test if its alphanumeric.

Filtering with 1 character:
image

Min char of 3:
image

Max char of 15:
image

Must be alphanumeric:
image

### CODE REVIEW

File: web/src/app/components/LoginComponents/UsernameField.tsx [hunk 1]
```diff
@@ -10,10 +10,21 @@ export default function UsernameField({username, updateUsername, closeModal} : U
...
+
         if (inputValue.trim().length < 3) {
             setHasUsernameError(true);
+            setErrorMsg('Name must be at least 3 characters')
+        } else if (inputValue.trim().length > 15) {
+            setHasUsernameError(true);
+            setErrorMsg('Name must be max 15 characters')
+        } else if (validCharacterRegex.test(inputValue.trim())) {
+            setHasUsernameError(true);
+            setErrorMsg('Name cannot have special characters')
```
Comment (jonathanplacatka) [inline web/src/app/components/LoginComponents/UsernameField.tsx]: change to 'Name cannot contain special characters'

File: web/src/app/multiplayer/page.tsx [hunk 2]
```diff
@@ -25,11 +25,10 @@ export default function Multiplayer() {
 
     const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
         
-        //TODO: Maybe implement a better search function?? Kinda trash/wonky
-        const { value } = event.currentTarget;
+        const value = event.target.value;
         setSearch(value);
```
Comment (jonathanplacatka) [inline web/src/app/multiplayer/page.tsx]: I think we can delete this search state variable now right?

Comment (YingLiang2) [reply web/src/app/multiplayer/page.tsx]: I think so, will doublecheck.

Comment (YingLiang2) [reply web/src/app/multiplayer/page.tsx]: We don't need this but this also means we remove the search state variable on the TextInput which does look weird, but we don't need it so I guess we remove it.

Comment (jonathanplacatka) [reply web/src/app/multiplayer/page.tsx]: doesn't look weird, the default value for a textinput is just what the user typed in

**Review by jonathanplacatka — APPROVED** [review]

File: web/src/app/multiplayer/page.tsx (hunk 2 above)
Comment (YingLiang2) [reply web/src/app/multiplayer/page.tsx]: No I'm talking about on the textinput, it doesn't have a input={} value

**Pull request merged**